src/runtime.js

```diff
@@ -268,6 +269,17 @@ function adoptShared(global, callback) {
     setStyle(global.document.documentElement, 'cursor', 'pointer');
   }
 
+  // Some deferred polyfills.
+  if (
+    // eslint-disable-next-line no-undef
+    (INTERSECTION_OBSERVER_POLYFILL || getMode().localDev || getMode().test) &&
+    shouldLoadInObPolyfill(global)
+  ) {
+    Services.extensionsFor(global).preloadExtension(
+      'amp-intersection-observer-polyfill'
```

Comment:
Wonder if it is possible to have cache preload the extension based on user agent?

Reply:
I'd think so, yes. But does IE support preload?

Reply:
Or do you mean to actually proactively put the extension `<script>` on the page by the cache?

Reply:
I was thinking about `<link rel="preload'>` . I thought server may be able to detect the user agent and whether preload or native InOb is supported or not.
`<script>` would also work if the especially for the inabox case. My understanding is that you would always need to polyfill the InOb for inabox usecase? Is that right

Reply:
Why always? Only for IE and such, right?

Reply:
Ahhh, I missed the part that native InOb can handle the cross domain iframe case well. 
We don't need this polyfill plus the inabox host when native InOb is supported. 
Only for IE and such.